daemon/cluster/executor/container/controller.go

```diff
@@ -445,11 +441,10 @@ func (r *controller) Logs(ctx context.Context, publisher exec.LogPublisher, opti
 		return errors.Wrap(err, "container not ready for logs")
 	}
 
-	rc, err := r.adapter.logs(ctx, options)
+	msgs, err := r.adapter.logs(ctx, options)
```

Comment:
This should be something like:

```go
var cancel func()
ctx, cancel = context.WithCancel(ctx)
defer cancel()
```

...so that if we return an error at some point after this call, the goroutine spawned by `logs` won't leak, trying to write logs over the channel forever.